File: clients/javascript/mev-shield-client.js
/**
 * MEV Shield Node.js Client
 * Simple client for connecting to MEV Shield API
 */

const WebSocket = require('ws');
const axios = require('axios');

class MEVShieldClient {
    constructor(host = 'localhost', port = 8765) {
        this.baseUrl = `http://${host}:${port}`;
        this.wsUrl = `ws://${host}:${port}/ws`;
    }

    async getStatus() {
        try {
            const response = await axios.get(`${this.baseUrl}/status`);
            return response.data;
        } catch (error) {
            console.error('Status request failed:', error);
            return null;
        }
    }

    async getOpportunities() {
        try {
            const response = await axios.get(`${this.baseUrl}/opportunities`);
            return response.data;
        } catch (error) {
            console.error('Opportunities request failed:', error);
            return null;
        }
    }

    connectWebSocket() {
        return new Promise((resolve, reject) => {
            this.websocket = new WebSocket(this.wsUrl);
            
            this.websocket.on('open', () => {
                resolve(this.websocket);
            });
            
            this.websocket.on('error', (error) => {
                reject(error);
            });
        });
    }
}

module.exports = MEVShieldClient;

File: clients/python/mev_shield_client.py
"""
MEV Shield Python Client
Simple client for connecting to MEV Shield API
"""

import requests
import json
import asyncio
import websockets
from typing import Dict, List, Optional

class MEVShieldClient:
    def __init__(self, host: str = "localhost", port: int = 8765):
        self.base_url = f"http://{host}:{port}"
        self.ws_url = f"ws://{host}:{port}/ws"
    
    async def connect_websocket(self):
        """Connect to WebSocket for real-time updates"""
        try:
            self.websocket = await websockets.connect(self.ws_url)
            return self.websocket
        except Exception as e:
            print(f"WebSocket connection failed: {e}")
            return None
    
    def get_status(self):
        """Get server status"""
        try:
            response = requests.get(f"{self.base_url}/status")
            return response.json()
        except Exception as e:
            print(f"Status request failed: {e}")
            return None
    
    def get_opportunities(self):
        """Get current MEV opportunities"""
        try:
            response = requests.get(f"{self.base_url}/opportunities")
            return response.json()
        except Exception as e:
            print(f"Opportunities request failed: {e}")
            return None
